### DISCUSSION TITLE
Prompting Passphrase when Android device is connected...

### DISCUSSION THREAD

**Discussion by new-creation7** [andyholmes/gnome-shell-extension-gsconnect#1871] [Q&A]
Can someone please explain this popup (What's this Passphrase for) ?
![1727969891646](https://github.com/user-attachments/assets/fbfaaeb3-81f3-4d11-8d00-f6cbb5ce5d56)

Thank you

**Comment by new-creation7** [comment]
Is it related to **Mount**ing Android device's storage?

**Comment by vctls** [comment]
Also getting this on every login since I moved to Ubuntu 24.04. It's pretty annoying.
It may have something to do with the fact that I've disabled the keyring.

**Reply by ferdnyc** [reply]
Hmm.

If that's coming up at every login, I suspect it's triggered when we run `ssh-add` to add the SFTP key to your session's `ssh-agent`. IIUC, typically the agent uses a keyring that's unlocked by PAM when you log in. But if you've disabled some of that functionality, then I suppose you're seeing the results of it attempting to create/unlock a keystore.

It looks like the `ssh-add` is only run when the SFTP plugin attempts to actually _**mount**_ a remote share, though. So, turning off the "SFTP Automount" option in the paired device(s)' **Advanced > Experimental** settings should be enough to prevent it unless you explicitly request a mount.

**Reply by vctls** [reply]
Not sure it's connected, but I've set up an `ssh-agent` service manually after disabling the Gnome keyring, and since then it hasn't complained. SFTP Automount is still enabled in the settings (I don't think it's actually working, but that's another issue).

**Reply by ferdnyc** [reply]
\*nod\* SFTP has other issues completely unrelated to key management. But, as long as an `ssh-agent` is running that give the plugin what it needs for that part.
